mubench.pipeline/data/build_command.py

```diff
@@ -0,0 +1,190 @@
+        shutil.copy(os.path.join(os.path.dirname(__file__), 'classpath.gradle'), os.path.join(project_dir, buildfile_dir))
+        command = "gradle :printClasspath -b '{}'".format(os.path.join(buildfile_dir, "classpath.gradle"))
+        output = Shell.exec(command, cwd=project_dir, logger=self.logger)
+        dependencies = GradleCommand.__parse_gradle_classpath(output)
+        _copy_classpath(dependencies, dep_dir, compile_base_path)
+
+    @staticmethod
+    def __parse_gradle_classpath(shell_output: str) -> Set[str]:
+        # shell_output looks like:
+        # :printClasspath
+        # /path/dependency1.jar
+        # /path/dependency2.jar
+        #
+        # BUILD SUCCESSFUL
+
+        lines = shell_output.splitlines()
+        first_dependency_idx = next(i for i, line in enumerate(lines) if line == ":printClasspath") + 1
+        first_empty_line_idx = next(i for i, line in enumerate(lines) if not line)
+        return set(lines[first_dependency_idx:first_empty_line_idx])
+
+    @staticmethod
+    def __parse_buildfile_dir(args):
+        buildfile_dir = ""
+
+        if "-p" in args:
+            buildfile_dir = args[args.index("-p") + 1]
+        elif "--project-dir" in args:
+            buildfile_dir = args[args.index("--project-dir") + 1]
+
+        return buildfile_dir
+
+
+class AntCommand(BuildCommand):
+    @staticmethod
+    def name() -> str:
+        return "ant"
+
+    def _extend_args(self, args: List[str]) -> List[str]:
+        return ["-debug", "-verbose"] + args
+
+    def _copy_dependencies(self, exec_output: str, project_dir: str,
+                           dep_dir: str, compile_base_path: str) -> None:
+        dependencies = AntCommand.__parse_ant_classpath(exec_output)
+        _copy_classpath(dependencies, dep_dir, compile_base_path)
+
+    @staticmethod
+    def __parse_ant_classpath(shell_output: str) -> Set[str]:
+        # shell_output looks like:
+        #   [javac] '-classpath'
+        #   [javac] '/project/build:/path/dep1.jar:/path/dep2.jar'
+
+        classpath = set()
+        lines = shell_output.splitlines()
+        for line in [lines[i + 1] for i, line in enumerate(lines) if "[javac] '-classpath'" in line]:
+            if line:
+                classpath_start_idx = line.find("'") + 1
+                classpath_end_idx = line.find("'", classpath_start_idx)
+                classpath.update(line[classpath_start_idx:classpath_end_idx].split(":"))
+
+        return classpath
+
+
+def _copy_classpath(dependencies: Set[str], dep_dir: str, compile_base_path: str):
+    remove_tree(dep_dir)
+    makedirs(dep_dir, exist_ok=True)
+    for dependency in dependencies:
+        if os.path.isdir(dependency):
+            # dependency is a classes directory
+            dep_name = os.path.relpath(dependency, compile_base_path)
+            dep_name = dep_name.replace(os.sep, '-')
+            _create_jar(dependency, os.path.join(dep_dir, dep_name + ".jar"))
```

Comment:
This class starts with a single `_` while the function declared below starts with two `__`!